andela/ah-backend-prime
# #164798295 Enable a user comment on an article

#### What does this PR do?
Ensures users comment on Articles

#### Description of Task to be completed?
- Write tests for model creation
- Create models for comments
- Create models for replies to comments
- Add Urls for commment manipulation
- Add view classes  for comment and reply manipulation

#### How should this be manually tested?
- Run the server
- Connect to swagger docs using this link
`localhost:8000/swagger-docs/`

Enter this data in the endpoint
- api/v1/articles/comments/{slug}/

```
{
	"body":"This is my comment"
}
```

Expect a result similar to this
```
{
    "comment": {
        "id": 4,
        "author": {
            "username": "patritzw",
            "bio": "",
            "email": "patrick.oakls@andeqsdla.com",
            "full_name": "",
            "image": "",
            "followers_no": 0,
            "following_no": 0
        },
        "article": {
            "author": {
                "username": "patritzw",
                "bio": "",
                "email": "patrick.oakls@andeqsdla.com",
                "full_name": "",
                "image": "",
                "followers_no": 0,
                "following_no": 0
            },
            "title": "Cedes",
            "average_rating": 0,
            "description": "You jus adsaffaawonaseat",
            "body": "Oh really, I didsassa, ifdafsds that asfawetrue",
            "createdAt": "2019-04-15T12:08:37.101772Z",
            "updatedAt": "2019-04-15T12:08:37.101791Z",
            "slug": "cedes-0x3",
            "tagList": []
        },
        "body": "Na thats some sht",
        "parentId": 0,
        "createdAt": "2019-04-15T12:23:14.056027Z",
        "updatedAt": "2019-04-15T12:23:14.056087Z"
    }
}
```
 #### Relevant Pivotal Tracker story
[#164798295](https://www.pivotaltracker.com/story/show/164798295)
#### Screen shots
<img width="1417" alt="Screenshot 2019-04-15 at 17 26 30" src="https://user-images.githubusercontent.com/43245704/56140474-abe91d00-5fa3-11e9-962a-f417b96e11ac.png">
